tsdb/head.go

```diff
@@ -1420,6 +1425,7 @@ func (h *Head) Close() error {
 	if h.wal != nil {
 		merr.Add(h.wal.Close())
 	}
+	h.closed = true
```

Comment:
It would be safe to mark this in the beginning so that there is no race in accessing the ones that are already closed (h.chunkDiskMapper and h.wal).